packages/truffle-decode-utils/src/conversion.ts

```diff
@@ -146,4 +146,77 @@ export namespace Conversion {
       return value;
     }
   }
-}
\ No newline at end of file
+
+  /**
+   * Convert a mapping representation into a JS Map
+   *
+   * Only converts integer types to BN right now, leaving other keys else alone
+   */
+  export function toMap(
+    { keyType, members }: any,
+    convertValue?: (value: any) => any
```

Comment:
Hm, I'm a little confused what the point of this generality here is, with `convertValue` passed as an (optional) input parameter.  Seems to me it's only ever called with `convertValue` equal to `cleanMappings`.  This isn't wrong, but it's a little confusing and I think over-complicated.  I mean, leave it in if you have some reason for it, but otherwise I'd recommend removing `convertValue` and just invoking `cleanMappings`.